assets/js/base/context/cart-checkout/payment-methods/use-payment-method-registration.js

```diff
@@ -69,80 +95,57 @@ const usePaymentMethodRegistration = (
 		};
 	}, [ cartTotals, cartNeedsShipping, shippingAddress ] );
 
-	// Handles initialization of all payment methods.
+	// Initialize payment methods on mount.
 	// Note: registeredPaymentMethods is not a dependency because this will not
 	// change in the life of the hook, it comes from an externally set value.
 	useEffect( () => {
-		// if all payment methods are initialized then bail.
-		if ( isInitialized ) {
-			return;
-		}
-		const updatePaymentMethod = ( current, canPay = true ) => {
-			if ( canPay ) {
-				setInitializedPaymentMethod( current );
-			}
-			// update the initialized count
-			countPaymentMethodsInitializing.current--;
-			// if count remaining less than 1, then set initialized.
-			if ( countPaymentMethodsInitializing.current < 1 ) {
-				setIsInitialized( true );
-			}
-		};
-		// loop through payment methods and see what the state is
 		for ( const paymentMethodName in registeredPaymentMethods ) {
 			const current = registeredPaymentMethods[ paymentMethodName ];
-			// if in editor context then we bypass can pay check.
+
 			if ( isEditor ) {
-				updatePaymentMethod( current );
-			} else {
-				Promise.resolve(
-					current.canMakePayment( canPayArgument.current )
-				)
-					.then( ( canPay ) => {
-						if ( canPay.error ) {
-							throw new Error( canPay.error.message );
-						} else {
-							updatePaymentMethod( current, canPay );
-						}
-					} )
-					.catch( ( error ) => {
-						updatePaymentMethod( current, false );
-						if ( CURRENT_USER_IS_ADMIN ) {
-							throw new Error(
-								sprintf(
-									__(
-										// translators: %s is the error method returned by the payment method.
-										'Problem with payment method initialization: %s',
-										'woo-gutenberg-products-block'
-									),
-									error.message
-								)
-							);
-						}
-					} );
+				setInitializedPaymentMethod( current );
+				continue;
 			}
+
+			Promise.resolve( current.canMakePayment( canPayArgument.current ) )
+				.then( ( canPay ) => {
+					if ( canPay.error ) {
+						throw new Error( canPay.error.message );
+					}
+					setInitializedPaymentMethod( current, canPay );
+				} )
+				.catch( handleRegistrationError );
 		}
-	}, [ isInitialized, isEditor ] );
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
There were comments specifically in place with regards to registeredPaymentMethods not being a dependency. I left this in place.